dmscripts/send_dos_opportunities_email.py

```diff
@@ -67,47 +95,122 @@ def get_html_content(briefs, number_of_days):
     return {"html": html_content}
 
 
-def main(data_api_client, mailchimp_client, lot_data, number_of_days, framework_slug):
-    logger.info(
-        "Begin process to send DOS notification emails for '{0}' lot".format(lot_data["lot_slug"]),
-        extra={"lot_data": lot_data, "number_of_days": number_of_days}
-    )
+def get_live_briefs_by_framework_and_lot(client, start_date, end_date):
+    """
+    Create a map of all live briefs, organised by framework and lot.
+    A framework/lot will only be included if there are live briefs for it in the time range given.
+
+    :param client: Data API client
+    :param start_date: YYYY-MM-DD
+    :param end_date: YYYY-MM-DD
+    :return: dict of live briefs, e.g.
+        {
+            'digital-outcomes-and-specialists-2': {
+                'digital-outcomes': [brief1, brief2, ...],
+                'user-research-participants': [brief3]
+            },
+            'digital-outcomes-and-specialists-3': {
+                'digital-specialists': [brief4, brief5],
+                'digital-outcomes': [brief6, brief7]
+            }
+        }
+    """
+    briefs = get_live_briefs_between_two_dates(client, start_date, end_date)
+    briefs_by_fw_and_lot = {}
 
+    for brief in briefs:
+        brief_fw_slug = brief['frameworkSlug']
+        brief_lot_slug = brief['lotSlug']
+        if brief['frameworkSlug'] in briefs_by_fw_and_lot:
+            if brief_lot_slug in briefs_by_fw_and_lot[brief_fw_slug]:
+                briefs_by_fw_and_lot[brief_fw_slug][brief_lot_slug].append(brief)
+            else:
+                briefs_by_fw_and_lot[brief_fw_slug][brief_lot_slug] = [brief]
+        else:
+            briefs_by_fw_and_lot[brief_fw_slug] = {
+                brief_lot_slug: [brief]
+            }
+    return briefs_by_fw_and_lot
+
+
+def main(data_api_client, mailchimp_client, number_of_days, framework_override, list_id_override, lot_slug_override):
```

Comment:
I'd suggest making some of these arguments keyword only

```suggestion
def main(data_api_client, mailchimp_client, number_of_days, *, framework_override, list_id_override, lot_slug_override):
```

Reply:
Good call. I held off on that originally because I "didn't want to change too much" 🤦‍♀ Anyway,  will add a commit